src/blockchain/test.rs

```diff
@@ -49,6 +50,106 @@ fn test_txs_cant_be_duplicated() -> Result<(), BlockchainError> {
     Ok(())
 }
 
+#[test]
+fn test_cant_apply_unsigned_tx() -> Result<(), BlockchainError> {
+    let miner = Wallet::new(Vec::from("MINER"));
+    let alice = Wallet::new(Vec::from("ABC"));
+    let bob = Wallet::new(Vec::from("CBA"));
+
+    let mut genesis_block = genesis::get_test_genesis_block();
+    genesis_block.header.proof_of_work.target = 0x00ffffff;
+    genesis_block.body = vec![Transaction {
+        src: Address::Treasury,
+        data: TransactionData::RegularSend {
+            dst: alice.get_address(),
+            amount: 10_000,
+        },
+        nonce: 1,
+        fee: 0,
+        sig: Signature::Unsigned,
+    }];
+
+    let mut chain = KvStoreChain::new(db::RamKvStore::new(), genesis_block.clone())?;
+
+    // Create unsigned signed tx
+    let unsigned_tx = Transaction {
+        src: alice.get_address(),
+        data: TransactionData::RegularSend { dst: bob.get_address(), amount: 1000 },
+        nonce: 1,
+        fee: 300,
+        sig: Signature::Unsigned,
+    };
+    let unsigned_tx = TransactionAndDelta {
+        tx: unsigned_tx,
+        state_delta: None,
+    };
+
+    // Ensure apply_tx will raise
+    match chain.apply_tx(&unsigned_tx.tx, false) {
+        Ok(_) => assert!(false, "Unsigned transaction shall not be applied"),
+        Err(e) => assert_eq!(e.to_string(), BlockchainError::SignatureError.to_string())
```

Comment:
Why compare strings? Can we use `matches!` macro instead?

Reply:
yep I can do that. I thought like this it will be more minimal

Reply:
should be fine now